### PR TITLE
Fix wrong construction of ElementaryTypeNameToken

### PR DESCRIPTION
**Pull request by Marenz**
And add an assert to prevent it in the future

Split off #7153.

### CODE REVIEW

File: liblangutil/Token.cpp [hunk 1]
```diff
@@ -72,6 +72,9 @@ void ElementaryTypeNameToken::assertDetails(Token _baseType, unsigned const& _fi
 			"No elementary type " + string(TokenTraits::toString(_baseType)) + to_string(_first) + "x" + to_string(_second) + "."
 		);
 	}
+	else if (_baseType == Token::Address)
```
Comment (chriseth) [inline liblangutil/Token.cpp]: What about
```suggestion
	else
```
?

Comment (Marenz) [reply liblangutil/Token.cpp]: testing the change

File: liblangutil/Token.cpp [hunk 2]
```diff
@@ -72,6 +72,9 @@ void ElementaryTypeNameToken::assertDetails(Token _baseType, unsigned const& _fi
 			"No elementary type " + string(TokenTraits::toString(_baseType)) + to_string(_first) + "x" + to_string(_second) + "."
 		);
 	}
+	else
+		solAssert(_first == 0 && _second == 0, "Address should not have size arguments");
```
Comment (axic) [inline liblangutil/Token.cpp]: The comment seems to be wrong?

Comment (Marenz) [reply liblangutil/Token.cpp]: updated.
@chriseth why is it okay to expect 0 for both here?

**Review by chriseth — APPROVED** [review]

**Pull request merged**